src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -340,13 +339,22 @@ func (cfg Config) cloudPollSleepTime() time.Duration {
 	return time.Duration((CloudConfigPollInterval.Nanoseconds() / 2) + rand.Int63n(CloudConfigPollInterval.Nanoseconds()))
 }
 
-func (cfg Config) fetchCloudConfig() ([]byte, error) {
-	url := cfg.CloudConfig
+// updateFrom creates a new Config by 'merging' the given yaml into this Config.
+// The masquerade sets, the collections of servers, and the trusted CAs in the
+// update yaml  completely replace the ones in the original Config.
+func (updated *Config) updateFrom(newCfg *Config) error {
+	// XXX: does this need a mutex, along with everyone that uses the config?
+	return deepcopy.Copy(updated, newCfg)
```

Comment:
So you're saying this will preserve things like user settings that aren't in the cloud config right @fffw?